PhysicsTools/NanoAOD/plugins/BToKsteeProducer.cc

```diff
@@ -0,0 +1,790 @@
+							  int(rint(refitVertexBToKstEE->degreesOfFreedom())));
+
+                    
+		      double cosAlpha = computeCosAlpha(refitBToKstEE,refitVertexBToKstEE,beamSpot);
+                    
+		      double mass_err = sqrt(refitBToKstEE->currentState().kinematicParametersError().matrix()(6,6));              
+      
+		      pat::CompositeCandidate BToKstEECand;
+		      BToKstEECand.addDaughter( ele1 , "ele1");
+		      BToKstEECand.addDaughter( ele2 , "ele2");
+		      BToKstEECand.addDaughter( kaon, "kaon");
+		      BToKstEECand.addDaughter( pion, "pion");
+
+		      BToKstEECand.addUserInt("ele1_index", i);
+		      BToKstEECand.addUserInt("ele2_index", j);
+		      BToKstEECand.addUserInt("kaon_index", k);
+		      BToKstEECand.addUserInt("pion_index", l);
+		      
+		      math::XYZVector refitEle1V3D = refitEle1->refittedTransientTrack().track().momentum();
+		      BToKstEECand.addUserFloat("ele1_pt",     sqrt(refitEle1V3D.perp2()));
+		      BToKstEECand.addUserFloat("ele1_eta",    refitEle1V3D.eta());
+		      BToKstEECand.addUserFloat("ele1_phi",    refitEle1V3D.phi());
+		      BToKstEECand.addUserFloat("ele1_charge", refitEle1->currentState().particleCharge());
+
+		      math::XYZVector refitEle2V3D = refitEle2->refittedTransientTrack().track().momentum();
+		      BToKstEECand.addUserFloat("ele2_pt",     sqrt(refitEle2V3D.perp2()));
+		      BToKstEECand.addUserFloat("ele2_eta",    refitEle2V3D.eta());
+		      BToKstEECand.addUserFloat("ele2_phi",    refitEle2V3D.phi());
+		      BToKstEECand.addUserFloat("ele2_charge", refitEle2->currentState().particleCharge());		    
+
+		      TLorentzVector ele1cand;
+		      ele1cand.SetPtEtaPhiM(sqrt(refitEle1V3D.perp2()), refitEle1V3D.eta(), refitEle1V3D.phi(), ElectronMass_);
+		      TLorentzVector ele2cand;
+		      ele2cand.SetPtEtaPhiM(sqrt(refitEle2V3D.perp2()), refitEle2V3D.eta(), refitEle2V3D.phi(), ElectronMass_);
+		      BToKstEECand.addUserFloat("eeKPiFit_e_mass", (ele1cand+ele2cand).Mag());
+
+		      math::XYZVector refitKaonV3D = refitKaon->refittedTransientTrack().track().momentum();
+		      BToKstEECand.addUserFloat("kaon_pt",    sqrt(refitKaonV3D.perp2()));
+		      BToKstEECand.addUserFloat("kaon_eta",   refitKaonV3D.eta());
+		      BToKstEECand.addUserFloat("kaon_phi",   refitKaonV3D.phi());
+		      BToKstEECand.addUserFloat("kaon_charge",refitKaon->currentState().particleCharge());
+		      BToKstEECand.addUserFloat("kaon_DCASig", DCABS/DCABSErr);
+		      
+		      math::XYZVector refitPionV3D = refitPion->refittedTransientTrack().track().momentum();
+		      BToKstEECand.addUserFloat("pion_pt",    sqrt(refitPionV3D.perp2()));
+		      BToKstEECand.addUserFloat("pion_eta",   refitPionV3D.eta());
+		      BToKstEECand.addUserFloat("pion_phi",   refitPionV3D.phi());
+		      BToKstEECand.addUserFloat("pion_charge",refitPion->currentState().particleCharge());
+		      BToKstEECand.addUserFloat("pion_DCASig", DCABS/DCABSErr);
+
+		      TLorentzVector kaonCand;
```

Comment:
also here and below for the pionCand, I would use the mass of the refitKst outcome of the K pi vtx fit.